lib/modules/manager/bazel/update.ts

```diff
@@ -121,10 +121,6 @@ export async function updateDependency({
         result = updateCode(result, [idx, 'tag'], newValue);
       }
 
-      if (newDigest) {
-        result = updateCode(result, [idx, 'digest'], newDigest);
-      }
```

Comment:
Can this digest be changed independently of the value? or we assume it's immutably tied to the value/release?

Reply:
Well, according [documentation](https://github.com/bazelbuild/rules_docker/blob/master/docs/container.md#container_pull), both `tag` and `digest` are optional, so I think it's better to treat it as independently updatable.

But can we infer this knowledge from the `update` structure? For example, when `newValue !== currentValue`, or using some other criteria.

Reply:
If tag and digest are independently updateable, they should be updated as part of the autoreplace (Eventually). i.e. part of update, not artifacts

Artifacts should update things which are dependent on the value/digest

Reply:
How can I tell apart one case from another reliably?

From the current `update.ts` code, they're independently updatable, but I don't trust it. If I remove `tag` parameter from `WORKSPACE`, the `hash` value should be updated in `update.ts`. But when both `tag` and `hash` are present, I think `hash` depends on `tag` value.

I'm having problem with the the criteria itself.

Reply:
To me, the course of action seems like we need to see how Renovate creates update with and without `tag` value, so we can spot the difference and see which condition to use for deciding whether it's artifact upgrade or the regular one.

Reply:
Can the digest update without the tag changing? If so then it's part of update not artifacts (and should work with autoreplace)